### PR TITLE
[daint,dom] Update jupyterlab-1.1.1-CrayGNU-19.10-batchspawner.eb

### PR DESCRIPTION
**Pull request by twrobinson**
This PR adds ipyparaview and the jupyterlab extension. 
ipyparaview requires update of pillow to 7.0 or higher, so this package is updated.  
The ipyparaview labextension can not currently be installed from npm repo, nor from github, so requires a git clone and manual install.
The PR also add itkwidgets
This PR also fixes matplotlib widgets, which is currently not working (this PR adds specific versioning for jupyterlab extensions, using @ notation, to allow for a reproducible build)
The PR also adds the GPU monitoring dashboard extension from the RAPIDS team
The PR also modifies the postinstallcmds to be separated with && rahter than ;, and hence it catches failures in the npm build stage (these failures were not previously caught by easybuild) (thanks Eirini)
The PR also changes the base EasyBuild config to PythonBundle, so that it is guaranteed to fail if a package tries to download depenencies. This means all packages must be pinned (and included in order of their requirement), again improves reproducibility (thanks Eirini).
The PR also adds exts_default_options , which reduces the size of the recipe by about 4x (thanks Eirini)

### CODE REVIEW

**Comment by lucamar** [comment]
The test was successful @twrobinson, is this pull request ready for review?

**Comment by twrobinson** [comment]
Nope, it is work in progress.

**Comment by twrobinson** [comment]
retest this please

**Comment by twrobinson** [comment]
retest this please

**Comment by twrobinson** [comment]
Lab failing to build on daint-mc.

**Comment by twrobinson** [comment]
retest this please

**Comment by twrobinson** [comment]
retest this please

**Comment by twrobinson** [comment]
retest this please

**Comment by twrobinson** [comment]
retest this please

**Comment by twrobinson** [comment]
retest this please

**Comment by twrobinson** [comment]
retest this please

**Comment by twrobinson** [comment]
retest this please

**Comment by twrobinson** [comment]
retest this please

**Comment by twrobinson** [comment]
retest this please

**Comment by twrobinson** [comment]
retest this please

**Comment by twrobinson** [comment]
retest this please

**Comment by twrobinson** [comment]
retest this please

**Comment by twrobinson** [comment]
This PR is now ready for review

File: easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.1.1-CrayGNU-19.10-batchspawner.eb [hunk 1]
```diff
@@ -20,507 +14,200 @@ toolchain = {'name': 'CrayGNU', 'version': '19.10'}
...
-        'use_pip': True,
-        'source_urls': ['https://pypi.python.org/packages/source/b/bash_kernel/'],
-        }),
-     ('batchspawner', '68a1fcd', {
-        'req_py_majver': '3',
-        'req_py_minver': '6',
-        'use_pip': True,
+# batchspawner compute side cscript components
+    ('batchspawner', '68a1fcd', {
         'source_urls': ['https://github.com/jupyterhub/batchspawner/tarball/68a1fcd'],
```
Comment (rsarm) [inline easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.1.1-CrayGNU-19.10-batchspawner.eb:191]: here maybe use `%(version)s`?
```
'source_urls': ['https://github.com/jupyterhub/batchspawner/tarball/%(version)s'],
```

File: easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.1.1-CrayGNU-19.10-batchspawner.eb [hunk 2]
```diff
@@ -20,507 +14,200 @@ toolchain = {'name': 'CrayGNU', 'version': '19.10'}
...
-        'source_urls': ['https://pypi.python.org/packages/source/i/ipyparallel/'],
-        }),
-    ('docopt', '0.6.2', {
-        'req_py_majver': '3',
-        'req_py_minver': '6',
-        'use_pip': True,
-        'source_urls': ['https://pypi.python.org/packages/source/d/docopt/'],
+# ipyparaview, not yet available in pypi and npm registry, 24 April 2020
+        ('ipyparaview', 'cf1c6f4', {
+        'source_urls': ['https://github.com/NVIDIA/ipyparaview/tarball/cf1c6f4'],
```
Comment (rsarm) [inline easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.1.1-CrayGNU-19.10-batchspawner.eb:195]: same here with `%(version)s`?

File: easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.1.1-CrayGNU-19.10-batchspawner.eb (hunk 1 above)
Comment (twrobinson) [reply easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.1.1-CrayGNU-19.10-batchspawner.eb:191]: yes, thanks Rafael

File: easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.1.1-CrayGNU-19.10-batchspawner.eb (hunk 2 above)
Comment (twrobinson) [reply easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.1.1-CrayGNU-19.10-batchspawner.eb:195]: Yep

**Pull request merged**